cep-??.md

```diff
@@ -0,0 +1,134 @@
+
+### `name` and `prefix`
+
+Optional, `str`.
+
+Both fields refer to the _preferred_ name or path for the newly created environment. Tools SHOULD allow these suggestions to be overridden by the user with additional CLI flags or equivalent. If the proposed environment path exists, tools MUST NOT overwrite silently by default.
+
+Special names `base` and `root` SHOULD not be accepted. Prefixes targetting protected system paths SHOULD be rejected. Paths can contain tildes (`~`) and environment variables, and they MUST be expanded when present.
+
+The name of the environment (and the last component of the expanded`prefix` path) MUST NOT contain any of these characters: `/`, ` `, `:`, `#`.
+
+### `dependencies`
+
+Required, `list[str | dict[str, Any]]`.
+
+The simplest form for this section is a list of `str` encoding `MatchSpec`-compatible requirements. These items MUST be processed as conda requirements and submitted to the solver (along with `channels`) to obtain the contents for the resulting conda environment.
+
+This section can also contain "subsection" dictionaries that map `str` to arbitrary values. Each key refers to a non-conda installer tool that will process the associated contents as necessary. The additional subsections MUST be processed after the conda requirements. They SHOULD only add new contents. These keys SHOULD NOT overwrite existing contents.
+
+Currently known subsections include `pip`, the contents of which encode a list of `str` referring to PyPI requirements. How to process this list is left as an implementation detail, but common approaches involve invoking the `pip` command-line directly.
+
+Additional subsections are allowed. The conda client MUST error out if it cannot process unknown sections.
+
+### `channels`
+
+Optional, `list[str]`.
+
+These are the conda channels that will be queried to solve the requirements added in `dependencies`. They can be expressed with names, URLs and local paths.
+
+If not specified, the conda client MUST use the default configuration. When specified, these channels MUST be used to populate the channel list passed to the solver. Then default channel configuration MUST be appended, unless the special name `nodefaults` is present. When this is the case, the default configuration MUST no be appended. The `nodefaults` name MUST not be passed to the solver.
+
+### `variables`
+
+Optional, `dict[str, str]`.
+
+Keys MUST be valid environment variable names for the target operating system. We recommend sticking to names compatible with both Windows and POSIX standards. Values SHOULD be `str`. If they are not, they MUST be converted to `str`.
+
+These contents SHOULD be dumped into the `conda-meta/state` file of the target environment, or equivalent, so they can be set upon environment activation.
+
+## Examples
+
+Simplest possible `environment.yml`:
+
+```yaml
+dependencies:
+  - numpy
+```
+
+With a name:
+
+```yaml
+name: test
+dependencies:
+  - numpy >=1.10
```

Comment:
What about conditions if a dependency is used on different platforms, python versions, implementations, etc.?